org.adempiere.base/src/org/adempiere/model/MWlistboxCustomization.java

```diff
@@ -0,0 +1,138 @@
+											 int 			AD_User_ID, 
+											 String 		AD_WListboxName, 
+											 String 		trxName)
+	{
+		Query query = new Query(ctx, Table_Name, "AD_User_ID=? AND WListboxName=?", trxName);
+		return query.setClient_ID().setOnlyActiveRecords(true).setParameters(new Object[] { AD_User_ID, AD_WListboxName}).first();
+	}
+
+
+	private static ArrayList<String> cleanCustomization (String[] CustomizationNew)
+	{
+		for (int i = 0; i < CustomizationNew.length; i++ )
+		{
+			CustomizationNew[i] = CustomizationNew[i].substring(0, CustomizationNew[i].indexOf("=") );
+		}
+		return new ArrayList<String>(Arrays.asList(CustomizationNew)); 
+	}	
+
+	private static int columnExists(String 				searchColumnName, 
+								   ArrayList<String> 	columnList)
+	{
+		for (int i = 0; i < columnList.size(); i++)
+		{
+			if (columnList.get(i).toString().equals(searchColumnName))
+			{
+				return i;
+			}
+		}
+	return -1;
+	}
+
+	/**
+	 * Save the columnWidth of the columns of the WListBox 
+	 * @param ctx
+	 * @param AD_WListboxName
+	 * @param AD_User_ID
+	 * @param Custom
+	 * @param trxName
+	 * @return
+	 */
+	
+	public static boolean saveData(Properties 	ctx,  
+								   String 		AD_WListboxName, 
+								   int 			AD_User_ID, 
+								   String 		Custom, 
+								   String 		trxName) 
+	{	
+		MWlistboxCustomization WlistBoxCust = get(ctx, AD_User_ID, AD_WListboxName, trxName);
+	
+		if (WlistBoxCust != null && WlistBoxCust.getAD_Wlistbox_Customization_ID()> 0)
+		{
+			String[] custOrg = WlistBoxCust.getCustom().split(",");
+			ArrayList<String> newColumnList = new ArrayList<>( Arrays.asList(Custom.trim().split(",")) );
+			ArrayList<String> orgColumnList  = new ArrayList<>( Arrays.asList(WlistBoxCust.getCustom().trim().split(",")) );
+			//only run through this if the with has changed for a column
+			if (!orgColumnList.equals(newColumnList))
+			{
+				ArrayList<String> columnSearch = cleanCustomization (custOrg);
+				ArrayList<String> addColumn = new ArrayList<String>();
+				
+				for (int i = 0; i < newColumnList.size(); i++)
+				{
+					String searchColumnName = newColumnList.get(i).toString().substring(0,newColumnList.get(i).toString().indexOf("="));
+					int n = columnExists(searchColumnName, columnSearch);
+					if(n > -1)
+					{
+						orgColumnList.set(n, newColumnList.get(i).toString().stripLeading());
+					}
+					else
+					{ 
+						addColumn.add(newColumnList.get(i).toString().stripLeading());
+					}
+				}
+				if (addColumn.size() > 0)
+				{
+					orgColumnList.addAll(addColumn);
+				}
+				WlistBoxCust.setCustom(orgColumnList.toString().substring(1, orgColumnList.toString().length() -1).replaceAll("\\s", "") );
+			}
+		}
+		else
+		{
+			WlistBoxCust = new MWlistboxCustomization(ctx, 0, trxName); 
+			WlistBoxCust.setWlistboxName(AD_WListboxName);
+			WlistBoxCust.setAD_User_ID(AD_User_ID);
+			WlistBoxCust.setCustom(Custom);
+		}
+		if (Util.isEmpty(WlistBoxCust.getCustom(), true)) {
+			return WlistBoxCust.delete(true);
```

Comment:
It is a weird flow that we did all the process above for update/create and then only do the delete here.